### PR TITLE
Allow making non-native site links server-absolute

### PR DESCRIPTION
**Pull request by QuLogic**
... instead of globally absolute with an explicit server name. This allows the docs to be served from a different server (e.g., on localhost), without links leaking out to the real matplotlib.org.

### CODE REVIEW

**Comment by jklymak** [comment]
The top level won't resolve for the other websites that use the theme, though.  Ie mpl-third-party etc?

**Comment by QuLogic** [comment]
Locally maybe, but not once on the site. They are all on the same server. Maybe we need an option as in https://github.com/matplotlib/mpl-brochure-site/pull/41?

**Comment by QuLogic** [comment]
Alternatively, perhaps this should be `navbar_link_prefix`, defaulting to `https://matplotlib.org/stable/`, can be overridden for hte website with `/stable`, and setting `None` would mean fall back to `pathto`.

**Review by jklymak — APPROVED** [review]

File: docs/conf.py [hunk 1]
```diff
@@ -47,7 +47,12 @@
     # and CI builds https://github.com/pydata/pydata-sphinx-theme/pull/386
     "collapse_navigation": not is_release_build,
     "show_prev_next": False,
-    "native_site": False
+    # Determines the type of links produced in the navigation header:
+    # - absolute: Links point to the URL https://matplotlib.org/...
+    # - relative: Links point to top-level of the server /...
```
Comment (tacaswell) [inline docs/conf.py:52]: Discussed with Elliott on a a call, suggested this be rename to "server_stable" or something similar..

File: docs/conf.py [hunk 2]
```diff
@@ -51,7 +51,12 @@
     # and CI builds https://github.com/pydata/pydata-sphinx-theme/pull/386
     "collapse_navigation": not is_release_build,
     "show_prev_next": False,
-    "native_site": False
+    # Determines the type of links produced in the navigation header:
+    # - absolute: Links point to the URL https://matplotlib.org/...
+    # - server-stable: Links point to top-level of the server /...
```
Comment (tacaswell) [inline docs/conf.py:56]:
```suggestion
    # - server-stable: Links point to top-level of the server /stable/...
```

**Pull request merged**